modules/amf-event-monitor-web/src/main/java/com/liferay/registration/web/portlet/util/AuditEventUtil.java

```diff
@@ -0,0 +1,198 @@
+        return (int)auditEventList.stream().filter(auditEvent -> auditEvent.getEventType().equals(eventType)).count();
+    }
+
+    public static int getAuditEventsCountByUserId(long companyId, long userId, OrderByComparator orderByComparator) throws JSONException {
+        List<AuditEvent> auditEventList = getAuditEvents(companyId,0,getAuditEventsCount(companyId),orderByComparator);
+        return (int)auditEventList.stream().filter(auditEvent -> auditEvent.getUserId()==userId).count();
+    }
+
+    public static int getAuditEventsCountByTypeAndUserId(long companyId, String eventType, long userId, OrderByComparator orderByComparator) throws JSONException {
+        List<AuditEvent> auditEventList = getAuditEvents(companyId,0,getAuditEventsCount(companyId),orderByComparator);
+        return (int)auditEventList.stream().filter(auditEvent -> auditEvent.getEventType().equals(eventType)&& auditEvent.getUserId()==userId).count();
+    }
+
+    public static int getAuditEventsCount(
+            long companyId, long userId, String userName, Date createDateGT,
+            Date createDateLT, String eventType, String className, String classPK,
+            String clientHost, String clientIP, String serverName, int serverPort,
+            String sessionID, boolean andSearch) {
+
+        return _auditEventManager.getAuditEventsCount(
+                companyId, userId, userName, createDateGT, createDateLT, eventType,
+                className, classPK, clientHost, clientIP, serverName, serverPort,
+                sessionID, andSearch);
+    }
+
+    public static List<AuditEvent> getAuditEventsByEventType(long companyId, int start, int end,
+                                                             OrderByComparator orderByComparator, String eventType) throws JSONException {
+
+        List<AuditEvent> auditEventList = getAuditEvents(companyId,0,getAuditEventsCount(companyId),orderByComparator);
+        List<AuditEvent> auditEventListReturn = new ArrayList<>();
+
+        List<AuditEvent> auditEventListByType = auditEventList.stream().filter(
+                auditEvent -> auditEvent.getEventType().equals(eventType)
+        ).collect(Collectors.toList());
+
+        int listSize = auditEventListByType.size();
+
+        if(start>=listSize){
+            for(int auditEventIndex = 0; auditEventIndex<listSize;auditEventIndex++){
+                auditEventListReturn.add(auditEventListByType.get(auditEventIndex));
+            }
+        }
+        if(end>=listSize){
+            for(int auditEventIndex = start; auditEventIndex<listSize;auditEventIndex++){
+                auditEventListReturn.add(auditEventListByType.get(auditEventIndex));
+            }
+        } else {
+            for(int auditEventIndex = start; auditEventIndex<end;auditEventIndex++){
+                auditEventListReturn.add(auditEventListByType.get(auditEventIndex));
+            }
+        }
+
+        return auditEventListReturn;
+    }
```

Comment:
I think it should use a custom Finder to implement this.
Because the current implementation could cause a performance problem.
https://help.liferay.com/hc/en-us/articles/360030614252-Defining-a-Custom-Finder-Method